Mathlib/Analysis/Calculus/FDeriv/Star.lean

```diff
@@ -92,3 +100,28 @@ theorem fderivWithin_star (hxs : UniqueDiffWithinAt 𝕜 s x) :
 theorem fderiv_star :
     fderiv 𝕜 (fun y => star (f y)) x = ((starL' 𝕜 : F ≃L[𝕜] F) : F →L[𝕜] F) ∘L fderiv 𝕜 f x :=
   (starL' 𝕜 : F ≃L[𝕜] F).comp_fderiv
+
+end TrivialStar
+
+section NontrivialStar
+
+/-!
+## Composing on the left and right with `star`
+-/
+
+variable [StarAddMonoid E] [StarModule 𝕜 E] [ContinuousStar E] [NormedStarGroup 𝕜]
+
+/-- If `f` has derivative `f'` at `z`, then `star ∘ f ∘ star` has derivative `starL ∘ f' ∘ starL`
+at `star z`. -/
+lemma HasFDerivAt.star_star {f : E → F} {z : E} {f' : E →L[𝕜] F} (hf : HasFDerivAt f f' z) :
+    HasFDerivAt (star ∘ f ∘ star)
```

Comment:
Do we prefer writing `∘` or a `fun` here, for `HasFDerivAt` in general?

Reply:
`HasDerivAt.star` doesn't use composition, so perhaps `fun` is better here.

Reply:
Then it sounds like `HasDerivAt.star` is not following the convention established in the discussion at https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Convention.20for.20.60f.20*.20g.60.20and.20.2F.20or.20.60fun.20x.20.E2.86.A6.20f.20x.20*.20g.20x.60. If I understand correctly, the consensus of that discussion is that there is room in Mathlib for both the `fun` versions and the unapplied versions, but the versions with `fun` in the statement should have `fun` in the name.

Reply:
I think currently nothing conforms with the consensus there!

Reply:
Only `AnalyticAt` and its siblings (which figures, since Stefan K. was the one pushing for this change). 

Given the lack of consistency with that rule, can I suggest we leave this for another PR (possibly with accompanying zulip discussion)?